packages/application-components/src/experimental/components/product-picker-input/product-picker-input.spec.tsx

```diff
@@ -0,0 +1,216 @@
+import type { TRenderAppOptions } from '@commercetools-frontend/application-shell/test-utils';
+
+// import { graphql } from 'msw';
+// import { setupServer } from 'msw/node';
+import React from 'react';
+import {
+  renderApp,
+  fireEvent,
+  screen,
+} from '@commercetools-frontend/application-shell/test-utils';
+import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
+import ProductPickerInput from './product-picker-input';
+import FetchProductByPrefix from './fetch-product-by-prefix.ctp.graphql';
+import PrefetchProduct from './prefetch-selected-product.ctp.graphql';
+
+const productId = '6f69c022-b894-4deb-b19f-6335547cb5d7';
+const inputName = 'product-picker-input';
+
+const createTestProps = (custom = {}) => ({
+  name: inputName,
+  value: productId,
+  hasError: false,
+  isReadOnly: false,
+  isClearable: false,
+  isDisabled: false,
+  onChange: jest.fn().mockName('onChange'),
+  onInitialLoadError: jest.fn().mockName('onInitialLoadError'),
+  onLoadError: jest.fn().mockName('onLoadError'),
+  ...custom,
+});
+
+const createTestProduct = (custom = {}) => ({
+  id: productId,
+  key: 'milk-product',
+  masterData: {
+    current: {
+      nameAllLocales: [
+        {
+          locale: 'en',
+          value: 'Milk',
+        },
+      ],
+    },
+  },
+  ...custom,
+});
+
+const createPrefetchedProductQueryMock = (custom = {}) => ({
+  request: {
+    query: PrefetchProduct,
+    context: { target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM },
+    variables: { id: productId },
+  },
+  result: {
+    data: {
+      product: createTestProduct(),
+    },
+  },
+  ...custom,
+});
+
+const createPrefixedSearchProductQueryMock = (custom = {}) => ({
+  request: {
+    query: FetchProductByPrefix,
+    variables: {
+      where:
+        'masterData(current(name(en >= "H" and en < "I"))) or key >= "H" and key < "I"',
+    },
+  },
+  result: {
+    data: {
+      products: {
+        results: [createTestProduct()],
+      },
+    },
+  },
+  ...custom,
+});
+
+const render = (
+  ui: React.ReactElement,
+  renderOptions: Partial<TRenderAppOptions<{}>>
+) => {
+  return renderApp(
```

Comment:
Note: as follow up we should maybe look into making test-utils more generic available. For instance, I don't like much that we use the appshell test-utils in this package.

Reply:
yes I thought about that too. let's move this to an issue.